pager/src/main/java/com/google/accompanist/pager/SnappingFlingBehavior.kt

```diff
@@ -0,0 +1,253 @@
+}
+
+/**
+ * Create and remember a snapping [FlingBehavior] to be used with [LazyListState].
+ *
+ * @param lazyListState The [LazyListState] to update.
+ * @param decayAnimationSpec The decay animation spec to use for decayed flings.
+ * @param snapAnimationSpec The animation spec to use when snapping.
+ */
+@Composable
+fun rememberSnappingFlingBehavior(
+    lazyListState: LazyListState,
+    decayAnimationSpec: DecayAnimationSpec<Float> = rememberSplineBasedDecay(),
+    snapAnimationSpec: AnimationSpec<Float> = SnappingFlingBehaviorDefaults.snapAnimationSpec,
+): SnappingFlingBehavior = remember(lazyListState, decayAnimationSpec, snapAnimationSpec) {
+    SnappingFlingBehavior(
+        lazyListState = lazyListState,
+        decayAnimationSpec = decayAnimationSpec,
+        snapAnimationSpec = snapAnimationSpec,
+    )
+}
+
+/**
+ * A snapping [FlingBehavior] for [LazyListState]. Typically this would be created
+ * via [rememberSnappingFlingBehavior].
+ *
+ * @param lazyListState The [LazyListState] to update.
+ * @param decayAnimationSpec The decay animation spec to use for decayed flings.
+ * @param snapAnimationSpec The animation spec to use when snapping.
+ */
+class SnappingFlingBehavior(
+    private val lazyListState: LazyListState,
+    private val decayAnimationSpec: DecayAnimationSpec<Float>,
+    private val snapAnimationSpec: AnimationSpec<Float>,
+) : FlingBehavior {
+    /**
+     * The target page for any on-going animations.
+     */
+    var animationTarget: Int? by mutableStateOf(null)
+        private set
+
+    override suspend fun ScrollScope.performFling(
+        initialVelocity: Float
+    ): Float {
+        val pageInfo = currentLayoutPageInfo ?: return initialVelocity
+        val pageSize = pageInfo.size
+        val decayTargetOffset = decayAnimationSpec.calculateTargetValue(
+            initialValue = pageInfo.offset.toFloat(),
+            initialVelocity = initialVelocity
+        )
+
+        // If the decay fling will scroll more than the width of the page, fling with
+        // using a decaying scroll
+        return if (decayTargetOffset.absoluteValue >= pageSize) {
+            performDecayFling(initialVelocity, pageInfo)
+        } else {
+            // Otherwise we 'spring' to current/next page
+            performSpringFling(
+                index = when {
+                    // If the velocity is greater than 1 page per second (velocity is px/s), spring
+                    // in the relevant direction
+                    initialVelocity > pageSize -> {
+                        (pageInfo.index + 1).coerceAtMost(lazyListState.layoutInfo.totalItemsCount - 1)
+                    }
+                    initialVelocity < -pageSize -> pageInfo.index
+                    // If the offset exceeds the scroll threshold (in either direction), we want to
+                    // move to the next/previous item
+                    pageInfo.offset < -pageSize / 2 -> pageInfo.index + 1
```

Comment:
do we want to make sure we are no on the last item already, as you did few lines before

Reply:
Yep, because the velocity might be 0. 

This last 'fail-safe' check uses the position to determine the snap position.

Reply:
Added a comment to clarify this.